app/integration_action.go

```diff
@@ -140,6 +127,50 @@ func (a *App) DoPostActionWithCookie(postId, actionId, userId, selectedOption st
 		upstreamURL = action.Integration.URL
 	}
 
+	// get channel, user and team details
+	chanChan := make(chan store.StoreResult, 1)
```

Comment:
0/5 nit: chanChan is effectively synchronous (line 145 - waits for it) - maybe move the fetching of the user up to line #130 (as a goroutine) and then fetching the channel is simply synchronous?

Reply:
I moved the fetching of the user to line 79 in order to be asynchronous with the fetching of the post. 
1b85d42 adds more optimisations to the requests flow.